triton-constraint-circuit/src/lib.rs

```diff
@@ -775,6 +763,41 @@ impl<II: InputIndicator> ConstraintCircuitMonad<II> {
         (main_constraints, aux_constraints)
     }
 
+    fn substitute_node_and_produce_constraint(
```

Comment:
A function that does two things is bad design.[^and] That said, the entire degree lowering pipeline violates good software engineering practices in various places and needs an overhaul. If it is possible to have this function do one thing only, that design is preferred. If this is too difficult given the context the function lives in, that design flaw will have to be eliminated with the general overhaul of constraint building and degree lowering.

[^and]: Function names containing `_and_` are a red flag for this.

Reply:
This function has been renamed to `apply_substitution`.